velox/expression/CastExpr.cpp

```diff
@@ -689,6 +689,24 @@ void CastExpr::evalSpecialForm(
   auto fromType = inputs_[0]->type();
   auto toType = std::const_pointer_cast<const Type>(type_);
 
+  if (context.errors()) {
```

Comment:
This doesn't look right as we are still evaluating on failed rows if some rows have failures, but some don't.

Reply:
Indeed. Then we can use `MutableRemainingRows` defined in Expr.h to avoid copying when unnecessary.

Reply:
This works since we have `if (!remainingRows->hasSelections()) {` in the block that will check if there are rows that don't have errors.
The CI tests are passing as well.

Reply:
Never mind. I see what Masha meant after looking at `MutableRemainingRows`. I will use that. For a moment I felt the issue was related to correctness.

Reply:
Notice that we continue to evaluate CAST on all rows, even if some of them have errors already. We only skip evaluation if all rows have errors. We need to make sure that CAST is evaluated only on a subset of rows that do not have errors. That subset may or may not be empty.

Reply:
Masha, thanks for the clarification! I realized your intent after looking at the `MutableRemainingRows` API.
I addressed this issue.